src/basic/architecture.h

```diff
@@ -49,6 +49,54 @@ typedef enum {
 
 Architecture uname_architecture(void);
 
+/* CPU plus userspace ABI definitions. In most cases you'll want to use this, not Architecture, which is
+ * insufficient for every case apart from the very basics of comparing a running CPU with another. Never
+ * use Architecture when there is any connection with images, binaries, libraries, paths, etc. */
+typedef enum {
+        ABI_ALPHA,
+        ABI_ARC,
+        ABI_ARC_BE,
+        ABI_ARM,
+        ABI_ARMEL,
+        ABI_ARMHF,
+        ABI_ARM64,
+        ABI_ARM64_BE,
+        ABI_ARM_BE,
```

Comment:
So there's a "arm-el" and "arm-be" variants, but also "arm"? What does that mean?

Reply:
ARMEL is not little endian, it's software floating point. So it's ARM, ARMEL, ARMHF, and big endian counterparts. Two reasons to keep the old ARM as-is: no idea whether it should represent hf o el, so best to be left alone, and also it could represent the even older userspace ABI which was used until (I think) armv5, and it's called arm-linux-gnu in terms of gcc/glibc ABI. It's from older releases and no longer used. Given this could not have been realistically used for arm <= 7 until these changes, and if it was I have no idea for what, I think it's best to leave the old GPT GUIDs there, and use new ones for the more specific and correct ABIs.

Reply:
see https://wiki.debian.org/Multiarch/Tuples for the full list